crates/sc-consensus-subspace/src/block_import.rs

```diff
@@ -657,9 +660,16 @@ where
 
         let parent_weight = if block_number.is_one() {
             0
+        } else if let Some(parent_weight) =
+            aux_schema::load_block_weight(self.client.as_ref(), parent_hash)?
```

Comment:
This line should additionally account for the error:  `Parent block ... has no associated weight`

Reply:
In that case we get `None` and will be handled on line 672

Reply:
> In that case we get None and will be handled on line 672

This is correct. However, the snippet still contains an error: my comment originated from the failed test with commented code for writing zero weight for fast-sync. 

The debug showed that the condition of the grandparent doesn't hold because we import the state block bypassing the checks as well, this in turn highlighted the danger of using this technique when we have an option of "raw importing" blocks. Removing weight management from the method makes it error-prone because we have the option to insert blocks arbitrarily but add only a single specific case for bypassing absent weights.

I would also ask you to introduce improvements when you have the full code available for testing.

Reply:
Honestly I'm not sure I understand what you're talking about or where error is.

You're saying about some failed test, but what test has failed and how exactly?

Reply:
What whole solution? I do not understand you.